library/ssl_tls.c

```diff
@@ -6169,11 +6213,19 @@ void mbedtls_ssl_reset_checksum( mbedtls_ssl_context *ssl )
 #endif
 #if defined(MBEDTLS_SSL_PROTO_TLS1_2)
 #if defined(MBEDTLS_SHA256_C)
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+    psa_hash_setup( &ssl->handshake->fin_sha256_psa, PSA_ALG_SHA_256 );
```

Comment:
I think we should call `psa_hash_abort()` here before re-calling `psa_hash_setup()`.

Reply:
I think strictly speaking we must also add `psa_hash_init()` here.

Reply:
I'm not sure about that (documentation of psa_hash_abort):
`Once aborted, the operation object can be reused for another operation by calling psa_hash_setup() again.`

Reply:
Oh, great, that's clear then - Thanks!